### PR TITLE
chore(state): Append Api Versions flags to helm-diff

### PR DESCRIPTION
**Pull request by semoac**
Support for the missing `KubeVersion` flag on Helm-diff was added in this PR: https://github.com/databus23/helm-diff/pull/386.

Helm-diff will append these flags to `helm template`.

Changes:
 * Added unit test for `state.flagsForDiff`
 * Fix `appendApiVersionsFlags` to check `Helmstate` values if `ReleaseSpec` values aren't set.

### CODE REVIEW

**Comment by yxxhero** [comment]
@semoac Please add some unittest for this flag. Thanks very much.

**Comment by semoac** [comment]
> @semoac Please add some unittest for this flag. Thanks very much.

I added a unit test for those flags. Also, it seems like `appendApiVersionsFlags` wasn't handling  cases where `ApiVersions` and `KubeVersion` were set outside ReleaseSpec.  `HelmState.ExecuteTemplates` has a logic that mitigates the case, but by adding the inspection directly into `HelmState.appendApiVersionsFlags` the function now always fall back to `HelmState` values if they exist.

File: pkg/state/state.go [hunk 1]
```diff
@@ -2566,11 +2568,20 @@ func (st *HelmState) chartVersionFlags(release *ReleaseSpec) []string {
...
-		flags = append(flags, "--api-versions", a)
+
+	if len(r.ApiVersions) != 0 {
+		for _, a := range r.ApiVersions {
+			flags = append(flags, "--api-versions", a)
+		}
+	} else {
+		for _, a := range st.ApiVersions {
+			flags = append(flags, "--api-versions", a)
+		}
```
Comment (itscaro) [inline pkg/state/state.go:2579]:
```suggestion
    var apiVersions []string
	if len(r.ApiVersions) != 0 {
		apiVersions = r.ApiVersions
	} else {
		apiVersions = st.ApiVersions
	}
	for _, a := range apiVersions {
		flags = append(flags, "--api-versions", a)
	}
```

File: pkg/state/state.go [hunk 2]
```diff
@@ -2566,11 +2568,20 @@ func (st *HelmState) chartVersionFlags(release *ReleaseSpec) []string {
...
 	}
 
 	if r.KubeVersion != "" {
+		flags = append(flags, "--kube-version", r.KubeVersion)
+	} else if st.KubeVersion != "" {
 		flags = append(flags, "--kube-version", st.KubeVersion)
 	}
```
Comment (itscaro) [inline pkg/state/state.go:2586]:
```suggestion
	var kubeVersion string
	if r.KubeVersion != "" {
		kubeVersion = r.KubeVersion
	} else if st.KubeVersion != "" {
		kubeVersion = st.KubeVersion
	}
	flags = append(flags, "--kube-version", kubeVersion)
```

File: pkg/state/state.go (hunk 1 above)
Comment (itscaro) [reply pkg/state/state.go:2579]: maybe?

File: pkg/state/state.go (hunk 2 above)
Comment (itscaro) [reply pkg/state/state.go:2586]: maybe?

**Review by mumoshu — APPROVED** [review]
LGTM. Thanks for your contribution @semoac!
I'd appreciate it if you could submit a follow-up PR for comments made by @itscaro, if you could agree with those 🙏

**Pull request merged**